src/SignaturePad.js

```diff
@@ -173,6 +173,9 @@ class SignaturePad extends PureComponent {
     }
 
     scaleCanvas() {
+        // Avoid needlessly setting height/width if dimensions haven't changed
+        if (this._canvas && this._canvas.offsetWidth === this.canvasWidth && this._canvas.offsetHeight === this.canvasHeight) return;
+        
```

Comment:
There is no need to check for existence of `this._canvas` (see `componentDidMount()`) and it is not correct to check against `this._canvas.offsetWidth` and `this._canvas.offsetHeight`. The check should be done against the computed width and height variables, since those are the ones stored in `this.canvasWidth` and `this.canvasHeight`. So I think it needs to become
```javascript
...
const width = (this.props.width || this._canvas.offsetWidth) * ratio;
const height = (this.props.height || this._canvas.offsetHeight) * ratio;

if (width === this.canvasWidth && height === this.canvasHeight) return;
```